nvidia-bert/ort_addon/ort_supplement/ort_supplement.py

```diff
@@ -43,99 +50,94 @@ def setup_onnxruntime_with_mpi(args):
-
-    lr = get_lr(args, global_step, args.schedule)
-    learning_rate = torch.tensor([lr])
-    if args.fp16:
-        loss_scale = torch.tensor([args.ort_loss_scale.loss_scale_])
-        loss = model.train_step(input_ids, segment_ids, input_mask, masked_lm_labels, next_sentence_labels, learning_rate, loss_scale)
-        all_finite = 1
-        if isinstance(loss, (list, tuple)):
-            assert len(loss) == 2
-            loss, all_finite = loss
-    else:
-        loss = model(input_ids, segment_ids, input_mask, masked_lm_labels, next_sentence_labels, learning_rate)
+    model_desc = {
+        'inputs': [
+            ('input_ids', [args.train_batch_size, args.max_seq_length]),
+            ('segment_ids', [args.train_batch_size, args.max_seq_length]),
+            ('input_mask', [args.train_batch_size, args.max_seq_length]),
+            ('masked_lm_labels', [args.train_batch_size, args.max_seq_length]),
+            ('next_sentence_labels', [args.train_batch_size, 2])
+        ],
+        'outputs': [
+            ('loss', [], True)
+        ]
+    }
+
+    # TRAINING OPTIMIZER SPECIFICATION
+    param_optimizer = list(model.named_parameters())
+    no_decay = ['bias', 'gamma', 'beta', 'LayerNorm']
+    optim_config = optim.LambConfig(
+        lr=args.learning_rate, alpha=0.9, beta=0.999, lambda_coef=0.01, epsilon=1e-6,
+        do_bias_correction=True,
+        params = [{
+            'params' : [n for n, p in param_optimizer if any(nd in n for nd in no_decay)],
+            'alpha': 0.9, 'beta': 0.999, 'lambda_coef': 0.00, 'epsilon': 1e-6
+        }]
+    )
+
+    # LEARNING RATE SCHEDULE SPECIFICATION
+    lr_scheduler = optim.lr_scheduler.LinearWarmupLRScheduler(
+        total_steps=int(args.max_steps), warmup=args.warmup_proportion)
+
+    # DYNAMIC LOSS SCALING SPECIFICATION (mixed precision only)
+    loss_scaler = ort_amp.loss_scaler.DynamicLossScaler()
+
+    # ONNXRUNTIME TRAINER OPTIONS 
+    trainer_config = orttrainer.ORTTrainerOptions({
+        'device': {
+            'id': str(device), 
+            'mem_limit': int(args.gpu_memory_limit_gb * 1024 * 1024 *1024)
+        },
+        'batch': {
+            'gradient_accumulation_steps' : args.gradient_accumulation_steps
+        },
+        'distributed': {
+            'world_size': args.world_size,
+            'world_rank': args.world_rank,
+            'allreduce_post_accumulation': True if args.allreduce_post_accumulation else False,
+            'deepspeed_zero_optimization': {
+                'stage': 1 if args.deepspeed_zero_stage else 0,
+            }        
+        },
+        'lr_scheduler': lr_scheduler,
+        'mixed_precision': {
+            'enabled': True if args.fp16 else False,
+            'loss_scaler': loss_scaler if args.fp16 else None,
+        },
+        # OOM issue..
+        # 'debug': {
+        #      'deterministic_compute' : True
+        # },
+        '_internal_use': {
+            'onnx_opset_version': 12
+        }
```

Comment:
This is the default already and we want to hide internal stuff as much as possible